### ISSUE TITLE
Remove plone.app.iterate profile

### ISSUE THREAD

**Issue by mauritsvanrees** [plone/plone.app.iterate#99]
There is still a `plone.app.iterate:plone.app.iterate` profile, which is the original copy of the `default` profile.
It is referenced in the add-on list in CMFPlone, but I have [removed it](https://github.com/plone/Products.CMFPlone/pull/3311/commits/1128832b34521a36eda68d95a4d54bff27e605d2) in a pending PR. We should do the same in CMFPlone 5.2.

When migrating from Plone 5.0.2 or earlier, users may have `plone.app.iterate:plone.app.iterate` installed, instead of the default profile. We should have an upgrade step to fix this. Maybe just reuse the single existing upgrade step in this package, which reinstalls itself. Or we could do this in `plone.app.upgrade`, as it might get a bit confusing in here.

For backwards compatibility, it is probably best to keep the `plone.app.iterate` profile in Plone 5.2. But the upgrade step could already be done.

**Comment by gforcada** [comment]
Done, right?

**Comment by mauritsvanrees** [comment]
I checked by hacking the current profiles a bit, and it should be fine to remove the old profile name completely, also if the upgrade step in `plone.app.upgrade` to fix the confusion has not been run yet.

Specifically, if the `plone.app.iterate:plone.app.iterate` was installed, but is not available anymore, then the upgrade step works just fine:

- The code sees that the `plone.app.iterate:plone.app.iterate` profile is installed.
- It marks it as uninstalled.
- It applies the default profile of the package.

So should be safe to remove, but let's wait until 6.1.

I have just now created a project for Plone 6.1 and this is the first item on it:
https://github.com/orgs/plone/projects/25/views/1
